Source/Atomic/UI/UILayout.h

```diff
@@ -130,6 +130,17 @@ class ATOMIC_API UILayout : public UIWidget
     void SetLayoutDistribution(UI_LAYOUT_DISTRIBUTION distribution);
     void SetLayoutDistributionPosition(UI_LAYOUT_DISTRIBUTION_POSITION distribution_pos);
 
+    /// A different way of setting up a layout, using a series of characters to program the main 5 layout fields
+    /// character [0] = 'X' or 'Y'  for UI_AXIS = X(D), Y
+    /// character [1] = 'A'|'G'|'P' for UI_LAYOUT_SIZE = Available, Gravity(D), Perferred
+    /// character [2] = 'C'|'G'|'L'|'R'  for UI_LAYOUT_POSITION = Center(D), Gravity, LeftTop, RightBottom
+    /// character [3] = 'A'|'G'|'P'  for UI_LAYOUT_DISTRIBUTION = Available, Gravity, Perferred(D)
+    /// character [4] = 'C'|'L'|'R'  for UI_LAYOUT_DISTRIBUTION_POSITION, Center(D), LeftTop, RightBottom
+    /// A '-' character in any field will not program that entry.
+    /// Any character used that is not an allowed characters or a '-' will result in the default setting to be used.
+    /// Any text in the string above character 5 is ignored.
+    /// If the input string is less than 5 characters it will not program any of the fields.
+    void SetLayoutConfig ( const String &settings );
```

Comment:
This is really clever, reminds me of linux permissions, very 🐫 
![aws](https://user-images.githubusercontent.com/376203/27929452-eee32066-6247-11e7-85e2-6162154014cb.gif)